canvas_modules/common-canvas/src/common-properties/controls/multiselect/multiselect.jsx

```diff
@@ -112,11 +118,15 @@ class MultiSelectControl extends React.Component {
 		for (let i = 0; i < evt?.selectedItems?.length; i++) {
 			controlValues.push(evt.selectedItems[i].id);
 		}
+		if (!isEqual(this.selectedOptions, controlValues)) {
```

Comment:
Is this check actually needed?  Wouldn't the `handleOnChange` method only be called if values change?

Reply:
Carbon team also suggested to add this check -

> It looks like you could update your handleOnChange to compare the selectedItems that are in the event with your existing list, and only update the value if they have changed.

Removing this check causes infinite loop again. 

Infinite loop happens because -

1. We update `this.multiSelectDropdown = this.genSelectOptions(controlValues);` in `handleOnChange()`
2. We call `this.props.controller.updatePropertyValue()` in `handleOnChange()`
3. After the `updatePropertyValue` is complete, `componentDidUpdate()` is called in which we update `this.multiSelectDropdown = this.genSelectOptions(controlValues);` again
4. Because `this.multiSelectDropdown` is updated, again `handleOnChange()` method is called and Steps 1-4 keep happening in loop

Reply:
I see.  Thanks for the explanation.